src/test/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionSpec.scala

```diff
@@ -54,38 +54,37 @@ class SubmissionSpec(_system: ActorSystem) extends TestKit(_system) with FlatSpe
       val workspaceServiceConstructor = WorkspaceService.constructor(dataSource, workspaceDAO, entityDAO, methodConfigDAO, new HttpMethodRepoDAO(mockServer.mockServerBaseUrl), new HttpExecutionServiceDAO(mockServer.mockServerBaseUrl), MockGoogleCloudStorageDAO, submissionSupervisor, submissionDAO)_
       lazy val workspaceService: WorkspaceService = TestActorRef(WorkspaceService.props(workspaceServiceConstructor, userInfo)).underlyingActor
-      testCode(workspaceService)
-      submissionSupervisor ! PoisonPill
+      try {
+        testCode(workspaceService)
+      }
+      finally {
+        // for failed tests we also need to poison pill
+        submissionSupervisor ! PoisonPill
+      }
     }
   }
 
-  "Submission requests" should "400 when given an unparseable entity expression" in withWorkspaceService { workspaceService =>
-    val submissionRq = SubmissionRequest("dsde", "GoodMethodConfig", "Individual", "indiv1", Some("this.is."))
-    val rqComplete = workspaceService.createSubmission( testData.wsName, submissionRq ).asInstanceOf[RequestComplete[(StatusCode, String)]]
-    val (status, _) = rqComplete.response
-    assertResult(StatusCodes.BadRequest) {
-      status
-    }
-  }
-
-  it should "400 when given an entity expression that evaluates to an entity of the wrong type" in withWorkspaceService { workspaceService =>
-    val submissionRq = SubmissionRequest("dsde", "GoodMethodConfig", "PairSet", "ps1", Some("this.pairs"))
-    val rqComplete = workspaceService.createSubmission( testData.wsName, submissionRq ).asInstanceOf[RequestComplete[(StatusCode, String)]]
-    val (status, _) = rqComplete.response
-    assertResult(StatusCodes.BadRequest) {
-      status
+  private def checkSubmissionStatus(workspaceService:WorkspaceService, submissionId:String) = {
+    val submissionStatusRqComplete = workspaceService.getSubmissionStatus( testData.wsName, submissionId )
+
+    submissionStatusRqComplete match {
+      case RequestComplete((submissionStatus:StatusCode, submissionData:String)) => {
+        assertResult(StatusCodes.NotFound) {
+          submissionStatus
+        }
+        fail("Expected to get submission status but got 404 not found")
+      }
+      case RequestComplete(submissionData:Submission) => {
+        assertResult(submissionId) {
+          submissionData.submissionId
+        }
+      }
+      case _ => fail("Unable to get submission status")
     }
   }
 
-  it should "400 when given no entity expression and an entity of the wrong type" in withWorkspaceService { workspaceService =>
-    val submissionRq = SubmissionRequest("dsde", "GoodMethodConfig", "PairSet", "ps1", None)
-    val rqComplete = workspaceService.createSubmission( testData.wsName, submissionRq ).asInstanceOf[RequestComplete[(StatusCode, String)]]
-    val (status, _) = rqComplete.response
-    assertResult(StatusCodes.BadRequest) {
-      status
-    }
-  }
 
+  /* SUCCESS cases */
```

Comment:
You're right, and I don't know why they didn't. I'm still not a huge fan of splitting them into "success" and "fail" cases but I have bigger battles to fight than this one.